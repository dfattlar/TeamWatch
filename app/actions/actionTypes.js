"use strict";

export const START_WATCH = "START_WATCH";
export const STOP_WATCH = "STOP_WATCH";
export const TICK = "TICK";
export const RESET_ALL = "RESET_ALL";
export const RESET_TIME = "RESET_TIME";
export const OPEN_MODAL = "OPEN_MODAL";
export const CLOSE_MODAL = "CLOSE_MODAL";
export const NEW_ATHLETE_INPUT = "NEW_ATHLETE_INPUT";
export const ADD_ATHLETE = "ADD_ATHLETE";
export const ADD_ATHLETE_ERROR = "ADD_ATHLETE_ERROR";
export const ADD_SPLIT = "ADD_SPLIT";
export const MODE_CHANGE = "MODE_CHANGE";
export const RESET_ATHLETE_LIST = "RESET_ATHLETE_LIST";
export const ADD_HISTORY = "ADD_HISTORY";
export const REMOVE_ATHLETE_FROM_WATCH = "REMOVE_ATHLETE_FROM_WATCH";
export const ADD_ATHLETE_TO_WATCH = "ADD_ATHLETE_TO_WATCH";
export const DELETE_ATHLETE = "DELETE_ATHLETE";
export const DELETE_HISTORY = "DELETE_HISTORY";
