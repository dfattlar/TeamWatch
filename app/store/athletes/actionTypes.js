export const ADD_ATHLETE_TO_WATCH = 'ADD_ATHLETE_TO_WATCH';
export const REMOVE_ATHLETE_FROM_WATCH = 'REMOVE_ATHLETE_FROM_WATCH';
export const REMOVE_ALL_ATHLETES_FROM_WATCH = 'REMOVE_ALL_ATHLETES_FROM_WATCH';
export const NEW_ATHLETE_INPUT_FIRST = 'NEW_ATHLETE_INPUT_FIRST';
export const NEW_ATHLETE_INPUT_LAST = 'NEW_ATHLETE_INPUT_LAST';
export const CREATE_ATHLETE = 'CREATE_ATHLETE';
export const CREATE_ATHLETE_ERROR = 'CREATE_ATHLETE_ERROR';
export const RESET_ATHLETE_LIST = 'RESET_ATHLETE_LIST';
