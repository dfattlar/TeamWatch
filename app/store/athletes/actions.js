import * as types from './actionTypes';

export const addAthleteToWatch = (id, name) => {
  return dispatch => {
    dispatch(addingAthleteToWatch(id, name));
  };
};

export const removeAthleteFromWatch = id => {
  return dispatch => {
    dispatch(removingAthleteFromWatch(id));
  };
};

export const removeAllAthletesFromWatch = () => {
  return dispatch => {
    dispatch(removingAllAthletesFromWatch());
  };
};

export const createAthlete = (first, last) => {
  return dispatch => {
    dispatch(creatingAthlete(first, last));
  };
};

export const createAthleteError = error => {
  return dispatch => {
    dispatch(creatingAthleteError(error));
  };
};

const addingAthleteToWatch = (id, name) => ({
  type: types.ADD_ATHLETE_TO_WATCH,
  payload: {id, name},
});

const removingAthleteFromWatch = id => ({
  type: types.REMOVE_ATHLETE_FROM_WATCH,
  id,
});

const removingAllAthletesFromWatch = () => ({
  type: types.REMOVE_ALL_ATHLETES_FROM_WATCH,
});

const creatingAthlete = (first, last) => ({
  type: types.CREATE_ATHLETE,
  payload: {first, last},
});

const creatingAthleteError = error => ({
  type: types.CREATE_ATHLETE_ERROR,
  error,
});
