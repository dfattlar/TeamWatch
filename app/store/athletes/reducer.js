'use strict';

import * as types from './actionTypes';
import React from 'react';
import * as _ from 'lodash';

const initialState = {
  newAthleteInputFirst: '',
  newAthleteInputLast: '',
  addAthleteError: false,
  athleteStore: [],
};

export default function athletes(state = initialState, action = {}) {
  switch (action.type) {
    case types.CREATE_ATHLETE:
      const {first, last} = action.payload;
      let newAthlete = {
        id: Math.random()
          .toString(12)
          .substring(7),
        name: `${first} ${last}`,
        onWatch: false,
      };
      let arrUpdated = [...state.athletesStore, newAthlete];
      return {
        ...state,
        addAthleteError: false,
        athletesStore: arrUpdated,
        newAthleteInputFirst: '',
        newAthleteInputLast: '',
      };
    case types.ADD_ATHLETE_TO_WATCH:
      const updatedAthleteArr = state.athletesStore.map(function(athlete) {
        if (athlete.id === action.payload.id) {
          return {
            ...athlete,
            onWatch: true,
          };
        }
        return athlete;
      });

      return {
        ...state,
        athleteStore: updatedAthleteArr,
      };
    case types.REMOVE_ATHLETE_FROM_WATCH:
      const removeId = action.id;
      const updatedStore = state.athleteStore.map(athlete => {
        if (athlete.id === removeId) {
          return {
            ...athlete,
            onWatch: false,
          };
        }
        return athlete;
      });

      return {
        ...state,
        athleteStore: updatedStore,
      };

    case types.RESET_ATHLETE_LIST:
      let resetAthleteArr = state.athleteStore.map(function(athlete) {
        return {
          ...athlete,
          onWatch: false,
        };
      });
      return {
        ...state,
        athleteStore: resetAthleteArr,
      };

    // case types.DELETE_ATHLETE:
    //   const deleteId = action.id;
    //   const athleteIndex = state.athleteStore.findIndex(athlete => {
    //     return athlete.id === deleteId;
    //   });

    //   const deleteAthleteArr = [
    //     ...state.athleteStore.slice(0, athleteIndex),
    //     ...state.athleteStore.slice(athleteIndex + 1),
    //   ];

    //   return {
    //     ...state,
    //     athleteStore: deleteAthleteArr,
    //   };
    default:
      return state;
  }
}
