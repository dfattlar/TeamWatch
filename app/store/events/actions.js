import * as types from "./actionTypes";

export const addEvent = payload => {
  return dispatch => {
    dispatch(addingEvent(payload));
  };
};

const addingEvent = payload => ({
  type: types.ADD_EVENT,
  payload
});
