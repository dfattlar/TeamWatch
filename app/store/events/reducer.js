'use strict';

import * as types from './actionTypes';
import React from 'react';

const initialState = {
  eventsStore: [],
};

export default function event(state = initialState, action = {}) {
  switch (action.type) {
    case types.ADD_EVENT:
      const id = Math.random()
        .toString(12)
        .substring(4);
      const payloadWithId = {
        ...action.payload,
        id,
      };
      const updatedEvent = [...state.eventsStore, payloadWithId];
      return {
        ...state,
        eventsStore: updatedEvent,
      };
    case types.DELETE_EVENT:
      const deleteId = action.id;
      const eventIndex = state.eventsStore.findIndex(event => {
        return event.id === deleteId;
      });

      const deleteEventArr = [
        ...state.eventsStore.slice(0, eventIndex),
        ...state.eventsStore.slice(eventIndex + 1),
      ];

      return {
        ...state,
        eventsStore: deleteEventArr,
      };
    default:
      return state;
  }
}
