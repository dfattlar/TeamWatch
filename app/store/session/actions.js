import * as types from "./actionTypes";
import firebase from "react-native-firebase";

export const restoreSession = () => {
  return dispatch => {
    dispatch(sessionRestoring());

    let unsubscribe = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        dispatch(sessionSuccess(user));
        unsubscribe();
      } else {
        dispatch(sessionLogout());
        unsubscribe();
      }
    });
  };
};

export const loginUser = (email, password) => {
  return dispatch => {
    dispatch(sessionLoading());

    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .catch(error => {
        dispatch(sessionError(error.message));
      });

    let unsubscribe = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        dispatch(sessionSuccess(user));
        unsubscribe();
      }
    });
  };
};

export const signupUser = (email, password) => {
  return dispatch => {
    dispatch(sessionLoading());

    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .catch(error => {
        dispatch(sessionError(error.message));
      });

    let unsubscribe = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        dispatch(sessionSuccess(user));
        unsubscribe();
      }
    });
  };
};

export const logoutUser = () => {
  return dispatch => {
    dispatch(sessionLoading());

    firebase
      .auth()
      .signOut()
      .then(() => {
        dispatch(sessionLogout());
      })
      .catch(error => {
        dispatch(sessionError(error.message));
      });
  };
};

const sessionRestoring = () => ({
  type: types.SESSION_RESTORING
});

const sessionLoading = () => ({
  type: types.SESSION_LOADING
});

const sessionSuccess = user => ({
  type: types.SESSION_SUCCESS,
  user
});

const sessionError = error => ({
  type: types.SESSION_ERROR,
  error
});

const sessionLogout = () => ({
  type: types.SESSION_LOGOUT
});
