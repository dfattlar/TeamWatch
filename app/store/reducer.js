import {combineReducers} from 'redux';

import session from './session';
import watch from './watch';
import athletes from './athletes';
import events from './events';

export default combineReducers({
  session,
  watch,
  athletes,
  events,
});
