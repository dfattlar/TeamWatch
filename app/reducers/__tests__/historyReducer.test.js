import reducer from '../historyReducer';
import * as types from '../../actions/actionTypes';

const initialState = {
  historyStore: [],
};

describe('history reducer', () => {
  it('should handle ADD_HISTORY', () => {
    Math.random = () => 0.36232365838707015;
    expect(
      reducer(undefined, {
        type: types.ADD_HISTORY,
        payload: {watch: 'ok'},
      }),
    ).toMatchSnapshot();
  });

  it('should handle ADD_HISTORY test2', () => {
    Math.random = () => 0.36232365838707015;
    expect(
      reducer(
        {
          ...initialState,
          historyStore: [
            {
              athletesArray: [
                {
                  colorId: 1,
                  id: 'hfz5yu',
                  name: 'name1',
                  splits: [1, 2, 3],
                  totalTime: 6,
                },
              ],
              name: 'test1',
              relayFinishTime: 4122,
              startTime: 1499014397509,
            },
          ],
        },
        {
          type: types.ADD_HISTORY,
          payload: {
            athletesArray: [
              {
                colorId: 2,
                id: 'hfz5yu2',
                name: 'name2',
                splits: [1, 2, 3],
                totalTime: 6,
              },
            ],
            name: 'test2',
            relayFinishTime: 4122,
            startTime: 1499014397509,
          },
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle DELETE_HISTORY', () => {
    expect(
      reducer(
        {
          ...initialState,
          historyStore: [
            {
              watch: 'ok',
              id: '211879173a8639',
            },
          ],
        },
        {
          type: types.DELETE_HISTORY,
          id: '211879173a8639',
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle DELETE_HISTORY test2', () => {
    expect(
      reducer(
        {
          ...initialState,
          historyStore: [
            {
              watch: 'ok',
              id: '211879173a8639',
            },
            {
              watch: 'ok',
              id: '211879173a8638',
            },
            {
              watch: 'ok',
              id: '211879173a8637',
            },
          ],
        },
        {
          type: types.DELETE_HISTORY,
          id: '211879173a8638',
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle DELETE_HISTORY test3', () => {
    expect(
      reducer(
        {
          ...initialState,
          historyStore: [
            {
              watch: 'ok',
              id: '211879173a8639',
            },
            {
              watch: 'ok',
              id: '211879173a8638',
            },
            {
              watch: 'ok',
              id: '211879173a8637',
            },
          ],
        },
        {
          type: types.DELETE_HISTORY,
          id: '211879173a8637',
        },
      ),
    ).toMatchSnapshot();
  });
});
