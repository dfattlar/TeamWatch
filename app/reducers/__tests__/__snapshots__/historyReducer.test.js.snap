// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history reducer should handle ADD_HISTORY 1`] = `
Object {
  "historyStore": Array [
    Object {
      "id": "211879173a863",
      "watch": "ok",
    },
  ],
}
`;

exports[`history reducer should handle ADD_HISTORY test2 1`] = `
Object {
  "historyStore": Array [
    Object {
      "athletesArray": Array [
        Object {
          "colorId": 1,
          "id": "hfz5yu",
          "name": "name1",
          "splits": Array [
            1,
            2,
            3,
          ],
          "totalTime": 6,
        },
      ],
      "name": "test1",
      "relayFinishTime": 4122,
      "startTime": 1499014397509,
    },
    Object {
      "athletesArray": Array [
        Object {
          "colorId": 2,
          "id": "hfz5yu2",
          "name": "name2",
          "splits": Array [
            1,
            2,
            3,
          ],
          "totalTime": 6,
        },
      ],
      "id": "211879173a863",
      "name": "test2",
      "relayFinishTime": 4122,
      "startTime": 1499014397509,
    },
  ],
}
`;

exports[`history reducer should handle DELETE_HISTORY 1`] = `
Object {
  "historyStore": Array [],
}
`;

exports[`history reducer should handle DELETE_HISTORY test2 1`] = `
Object {
  "historyStore": Array [
    Object {
      "id": "211879173a8639",
      "watch": "ok",
    },
    Object {
      "id": "211879173a8637",
      "watch": "ok",
    },
  ],
}
`;

exports[`history reducer should handle DELETE_HISTORY test3 1`] = `
Object {
  "historyStore": Array [
    Object {
      "id": "211879173a8639",
      "watch": "ok",
    },
    Object {
      "id": "211879173a8638",
      "watch": "ok",
    },
  ],
}
`;
